src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -317,74 +313,39 @@ public void onGeolocationPermissionsShowPrompt(
 		});
 	}
 
+	public boolean getLocationPermissions() {
+		if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PERMISSION_GRANTED) {
+			return true;
+		}
+
+		String[] permissions = { Manifest.permission.ACCESS_FINE_LOCATION };
+		ActivityCompat.requestPermissions(this, permissions, ACCESS_FINE_LOCATION_PERMISSION_REQUEST);
+		return false;
+	}
+
+	@Override
+	public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+		if (requestCode != ACCESS_FINE_LOCATION_PERMISSION_REQUEST) {
+			return;
```

Comment:
I think it would be more correct for this to call `super.onRequestPermissionsResult`. In practice it probably makes no difference.

Reply:
I took the code from the example page, and they didn't call `super`: 
https://developer.android.com/training/permissions/requesting